src/views/workflow-history/workflow-history-timeline-reset-button/__tests__/workflow-history-timeline-reset-button.test.tsx

```diff
@@ -0,0 +1,190 @@
+import { type DescribeWorkflowResponse } from '@/route-handlers/describe-workflow/describe-workflow.types';
+import mockResolvedConfigValues from '@/utils/config/__fixtures__/resolved-config-values';
+import { mockResetActionConfig } from '@/views/workflow-actions/__fixtures__/workflow-actions-config';
+import { resetWorkflowActionConfig } from '@/views/workflow-actions/config/workflow-actions.config';
+import { describeWorkflowResponse } from '@/views/workflow-page/__fixtures__/describe-workflow-response';
+
+import WorkflowHistoryTimelineResetButton from '../workflow-history-timeline-reset-button';
+import type Props from '../workflow-history-timeline-reset-button.types';
+jest.mock('@/views/workflow-actions/config/workflow-actions.config', () => {
+  const originalModule = jest.requireActual(
+    '@/views/workflow-actions/config/workflow-actions.config'
+  );
+  return {
+    ...originalModule,
+    resetWorkflowActionConfig: {
+      ...mockResetActionConfig,
+      getRunnableStatus: jest.fn(),
+    },
+  };
+});
+
+describe('WorkflowHistoryTimelineResetButton', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('renders reset button when all conditions are met', async () => {
+    setup({});
+
+    // Wait for any async operations to complete
+    await screen.findByText('Reset');
+
+    const resetButton = screen.getByText('Reset');
+    expect(resetButton).toBeInTheDocument();
+  });
+
+  it('calls onReset when button is clicked', async () => {
+    const { mockOnReset } = setup({});
+
+    // Wait for any async operations to complete
+    const resetButton = await screen.findByText('Reset');
+    fireEvent.click(resetButton);
+
+    expect(mockOnReset).toHaveBeenCalledTimes(1);
+  });
+
+  it('does not render when reset action is not enabled', () => {
+    setup({
+      resetActionsEnabledConfig: 'DISABLED',
+    });
+
+    expect(screen.queryByText('Reset')).not.toBeInTheDocument();
+  });
+
+  it('does not render when reset is not runnable', () => {
+    setup({
+      isResetRunnable: false,
+    });
+
+    expect(screen.queryByText('Reset')).not.toBeInTheDocument();
+  });
+
+  it('does not render when workflow is loading', () => {
+    setup({
+      isWorkflowLoading: true,
+    });
+
+    expect(screen.queryByText('Reset')).not.toBeInTheDocument();
+  });
+
+  it('does not render when actions config is loading', () => {
+    setup({
+      isActionsEnabledLoading: true,
+    });
+
+    expect(screen.queryByText('Reset')).not.toBeInTheDocument();
+  });
+
+  it('does not render when workflow has error', () => {
+    setup({
+      isWorkflowError: true,
+    });
+
+    expect(screen.queryByText('Reset')).not.toBeInTheDocument();
+  });
+
+  it('does not render when actions config has error', () => {
+    setup({
+      isActionsEnabledError: true,
+    });
+
+    expect(screen.queryByText('Reset')).not.toBeInTheDocument();
+  });
+});
+
+function setup({
+  workflowId = 'test-workflow-id',
```

Comment:
Do we need some of these params? I don't see them being passed to the setup function above, so you may as well hardcode them inside